mne/epochs.py

```diff
@@ -662,7 +662,7 @@ def __next__(self, *args, **kwargs):
     def __hash__(self):
         if not self.preload:
             raise RuntimeError('Cannot hash epochs unless preloaded')
-        return object_hash(dict(info=self.info, data=self._data))
+        return object_hash(dict(info=hash(self.info), data=self._data))
```

Comment:
how about updating object_hash function rather than doing these changes in all objects?

you implement the hash method for Info so the object_hash function should be smart enough to use it.

that would remove changes in epochs.py, evoked.py etc.